### DISCUSSION TITLE
TCP or UDP include RSSI or signal strength value?

### DISCUSSION THREAD

**Discussion by SFSailor** [jvde-github/AIS-catcher#324] [Q&A] [answered]
I am looking at a system to evaluate ocean racing boats' AIS installations, some of which are terrible. I'd like to automagically collect signal strength data from a few stations. Will I be able to include RSSI in the stream that is sent out by UDP or TCP?  I assume the -o 5 option is the one I want....

**Comment by jvde-github** [comment, marked as answer]
The signal level is displayed as screen output with -o 3 or -o 5 (but -o 3 is sufficient). If you want to send in the same format over UDP use:

-u 127.0.0.1 5012 JSON on

as TCP Client:

-P 127.0.0.1 5012 JSON on

and as TCP server:

-S 5012 JSON on

Btw it is technically the signal level (and not signal to noise or something) so best to run with a fixed gain and not auto gain setting.

Does that help?

**Reply by SFSailor** [reply]
That is very helpful.  I was hoping it worked that way.  I now need to construct my own TCP or USP endpoint to collect the data. Since I want Lat/Lon, I guess the -o 5 option is best for me.  Thanks very much

**Reply by jvde-github** [reply]
well, you can use AIS-catcher as an end point as well. Suppose you are sending as TCP server (`-S 5012 JSON on`) you can read that again with `AIS-catcher -t txt ip-address-of-server 5012 -o 5`

If you are sending over udp (`-u ip-of-receiver 5012 JSON on`) you can read on the receiver with `AIS-catcher -x ip-of-receiver 5012 -o 5`)

So you do the full decoding on the receiving side

**Reply by SFSailor** [reply]
Jasper, yes that makes sense. My use case would have several receivers all sending data in, with the RSSI processed for distance from the sender, so I'd want to be sure to be able to tell the receivers apart.

In any case, I'd then need to pipe the decoded data into my own system for analysis.

First step will be to do it with just ONE of course, but at some point, I need my script to access the data.  I need to read the ReadMe about that.

**Reply by jvde-github** [reply]
If you run a receiver at some location listening to AIS messages, you can set a station ID, like:
```
AIS-catcher -d:0  -go station_id 12 -u 127.0.0.1 5012 json on
```
Then on the Receiving side I have:
```
AIS-catcher -x 127.0.0.1 5012 -o 5
```
It will include the station id in the JSON (this is both for -o 3 and -o 5), e.g.
```
{"class":"AIS","device":"AIS-catcher","version":0,"driver":0,"hardware":"","rxtime":"20241114202056","scaled":true,"channel":"A","nmea":["!AIVDM,1,1,,A,13bVm>0P00PCQ=FMksmf4?wh2<40,0*5A"],"signalpower":1024.000000,"ppm":1024.000000,"station_id":12,"type":1,"repeat":0,"mmsi":246003000,"status":0,"status_text":"Under way using engine","turn_unscaled":-128,"turn":-128,"speed":0.000000,"accuracy":true,"lon":4.263965,"lat":52.099342,"course":360.000000,"heading":511,"second":56,"maneuver":0,"raim":true,"radio":49408}
```
So you could set a station_id for each receiver and then when you receive and aggregate you have a unique station_id in there?

**Reply by SFSailor** [reply]
AHA. I was hoping that was the case, so I can use a single point of receipt.

**Reply by jvde-github** [reply]
Yes, though not sure it is fully documented in the readme yet.

**Reply by SFSailor** [reply]
Okay, with both -o 3 and -o 5 I get MMSI and RSSI, but with the -o 5 I was also hoping to get plaintext Lat/Lon as appears on the terminal output.  It is not, however, present in the UDP.  Is that as designed?

**Reply by jvde-github** [reply]
Convention in marine AIS is to send raw AIS NMEA over UDP. So we have the following equivalence between screen and UDP:

-o 1 <-> -u 127.0.0.1 5100
-o 3 <-> -u 127.0.0.1 5100 JSON ON
-o 5 <-> -u 127.0.0.1 5100 JSON_FULL on

Hope that helps

**Comment by SFSailor** [comment]
This is all working splendidly.

May I ask, how do I add -o 5 and JSON_FULL on to the json config file?  I've tried a few spots that did not work...
